src/tensor_ops/matmul/cpu_kernel.rs

```diff
@@ -235,13 +235,19 @@ impl<F: Dtype> super::MatMatBatch3Kernel<F> for Cpu
 where
     Self: MatMulImpl<F>,
 {
-    fn forward<const B: usize, M: Dim, const K: usize, N: Dim>(
+    fn forward<const B: usize, M: Dim, K: Dim, N: Dim>(
         &self,
-        lhs: &Self::Storage<(Const<B>, M, Const<K>), F>,
-        rhs: &Self::Storage<(Const<B>, Const<K>, N), F>,
+        lhs: &Self::Storage<(Const<B>, M, K), F>,
+        rhs: &Self::Storage<(Const<B>, K, N), F>,
     ) -> Result<Self::Storage<(Const<B>, M, N), F>, Self::Err> {
         let m: M = lhs.shape().1;
         let n: N = rhs.shape().2;
+
+        let k: K = lhs.shape().2;
+        let k2: K = rhs.shape().1;
+        if k != k2 {
+            return Err(CpuError::WrongNumElements);
+        }
```

Comment:
you can just do `assert_eq!` here instead of returning err. All the shape checks we have so far panic immediately instead of returning an err (the err is for memory allocation mainly)